lib/baseFactory.js

```diff
@@ -55,13 +55,21 @@ class BaseFactory {
             table: this.table,
             params: {
                 id,
-                data: config
+                data: {}
             }
         };
 
+        // Filter for valid keys
+        this.model.allKeys.forEach((key) => {
+            if (config[key]) {
+                modelConfig.params.data[key] = config[key];
+            }
+        });
+
         return nodeify.withContext(this.datastore, 'save', [modelConfig])
-            .then(modelData => this.createClass(hoek.applyToDefaults(modelData, {
-                datastore: this.datastore
+            .then(modelData => this.createClass(hoek.applyToDefaults(config, {
```

Comment:
yeah, why change this from `modelData` to `config` ? 

The base class will do essentially the same thing being done on lines 63 to 67, so you don't need all those extra "config" parameters anyway